app/controllers/api/v1/chats_controller.rb

```
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class Api::V1::ChatsController < ApplicationController
+  # rubocop:disable Metrics/PerceivedComplexity
+  def new_message
+    if message_params[:chat_id].present?
+      chat = Chat.find_by(id: message_params[:chat_id])
+      if chat.present?
+        if message_params[:user_id].present?
+          if check_user_id message_params[:user_id]
+            create_message message_params[:body], chat, message_params[:user_id]
+          else
+            render json: { error: 'non-valid user_id provided' }, status: :bad_request
+          end
+        else
+          create_message message_params[:body], chat
+        end
+      else
+        render json: { error: 'Chat not found' }, status: :bad_request
+      end
+    elsif message_params[:user_id].present?
+      render json: { error: 'you should provide a chat_id' }, status: :bad_request
+    else
+      chat = Chat.create
+      create_message message_params[:body], chat
+    end
+  end
+  # rubocop:enable Metrics/PerceivedComplexity
+
+  private
+
+  def create_message(body, chat, user_id = nil)
```

You should not create a chat. If the chat given has a parameter does not exists then the function should return a error message. The user from the widget has an endpoint function to create a chat